edit-post/store/effects.js

```diff
@@ -54,10 +54,21 @@ const effects = {
 		store.dispatch( setMetaBoxSavedData( dataPerLocation ) );
 
 		// Saving metaboxes when saving posts
+		const { isAutosavingPost } = select( 'core/editor' );
+		let shouldRequestMetaBoxUpdates = false;
 		subscribe( onChangeListener( select( 'core/editor' ).isSavingPost, ( isSavingPost ) => {
```

Comment:
> Maybe we could add support for watching multiple selectors in `onChangeListener` or just drop it and use custom logic.

What custom logic would you use?

Reply:
Do you think this custom logic is better than what I have? At first glance, what I have already seems more precise.

Reply:
It's not about better or worse, it's about the fact that we want to run the listener whenever the value of `isAutosavingPost`'s selector changes. In your code it's not the case, if for some reason (even if it's not possible at the moment) `isSavingPost` doesn't change but `isAutosavingPost` does, your listener is not being executed and I don't think it's intentional.

Reply:
`subscribe( () => {` fires for every change, which seems too frequent. Is there an alternative you'd suggest?

Reply:
There is none actually, we can bundle the multiple selectors support into `onChangeListener`. Maybe by supporting an "array of selectors" instead of a given "selector" as an argument passed to `onChangeListener` but in terms of performance, it doesn't change anything compared to my logic above.

That's how redux work in general, a unique `eventEmitter` calling all its listeners, It has proved to scale well in several contexts.

Reply:
Ok, updated in ed799682a